R/weight_draws.R

```diff
@@ -50,9 +52,12 @@ weight_draws <- function(x, weights, ...) {
 
 #' @rdname weight_draws
 #' @export
-weight_draws.draws_matrix <- function(x, weights, log = FALSE, ...) {
+weight_draws.draws_matrix <- function(x, weights, log = FALSE, pareto_smooth = FALSE, ...) {
   log <- as_one_logical(log)
   log_weights <- validate_weights(weights, x, log = log)
+  if (pareto_smooth) {
```

Comment:
can you make sure that all new input arguments are checked for validity?

Reply:
I added `checkmate::expect_flag(pareto_smooth)` in each of the methods. Would this be ok?

Reply:
Looks good!